
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      
      
      <link rel="icon" href="../../images/favicon_non_transparent.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.18">
    
    
      
        <title>Lecture 04 - Mathematical modelling in ecology and evolution (EEB314)</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.7e37652d.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


  
  
    
      
      
      
    
  
  
  <style>:root{.md-tag{--md-tag-icon:url('data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2024%2024%22%3E%3Cpath%20d%3D%22M5.5%207A1.5%201.5%200%200%201%204%205.5%201.5%201.5%200%200%201%205.5%204%201.5%201.5%200%200%201%207%205.5%201.5%201.5%200%200%201%205.5%207m15.91%204.58-9-9C12.05%202.22%2011.55%202%2011%202H4c-1.11%200-2%20.89-2%202v7c0%20.55.22%201.05.59%201.41l8.99%209c.37.36.87.59%201.42.59s1.05-.23%201.41-.59l7-7c.37-.36.59-.86.59-1.41%200-.56-.23-1.06-.59-1.42%22/%3E%3C/svg%3E');}}</style>

    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="purple">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#lecture-4-one-locus-selection" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Mathematical modelling in ecology and evolution (EEB314)" class="md-header__button md-logo" aria-label="Mathematical modelling in ecology and evolution (EEB314)" data-md-component="logo">
      
  <img src="../../images/favicon_non_transparent.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Mathematical modelling in ecology and evolution (EEB314)
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Lecture 04
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="purple"  aria-hidden="true"  type="radio" name="__palette" id="__palette_0">
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Overview

      </a>
    </li>
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../syllabus/instructors/" class="md-tabs__link">
          
  
  
  Syllabus

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../schedule/" class="md-tabs__link">
          
  
  
  Lectures

        </a>
      </li>
    
  

      
        
  
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../labs/schedule/" class="md-tabs__link">
          
  
  
  Labs

        </a>
      </li>
    
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../../acknowledgements/" class="md-tabs__link">
        
  
  
    
  
  Acknowledgements

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Mathematical modelling in ecology and evolution (EEB314)" class="md-nav__button md-logo" aria-label="Mathematical modelling in ecology and evolution (EEB314)" data-md-component="logo">
      
  <img src="../../images/favicon_non_transparent.svg" alt="logo">

    </a>
    Mathematical modelling in ecology and evolution (EEB314)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Syllabus
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Syllabus
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/instructors/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Instructors
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/when_and_where/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    When and where
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/course_structure/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Course structure
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/textbook/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Textbook
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/final_project/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Final Project
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/exams/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Exams
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/general_info/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    General info
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../syllabus/resources/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Other resources
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Lectures
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Lectures
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schedule
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../lecture-01/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lecture 1
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../lecture-02/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Lecture 2
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
      
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" >
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Labs
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            Labs
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../labs/schedule/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Schedule
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../labs/tutorial-01/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Tutorial 1
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../acknowledgements/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Acknowledgements
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#lecture-overview" class="md-nav__link">
    <span class="md-ellipsis">
      Lecture overview
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#1-review-of-hardy-weinberg" class="md-nav__link">
    <span class="md-ellipsis">
      1. Review of Hardy-Weinberg
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2-one-locus-haploid-selection-in-discrete-time" class="md-nav__link">
    <span class="md-ellipsis">
      2. One-locus haploid selection in discrete time
    </span>
  </a>
  
    <nav class="md-nav" aria-label="2. One-locus haploid selection in discrete time">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#life-cycle-diagram" class="md-nav__link">
    <span class="md-ellipsis">
      Life-cycle diagram
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deriving-the-equations" class="md-nav__link">
    <span class="md-ellipsis">
      Deriving the equations
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#simplifying" class="md-nav__link">
    <span class="md-ellipsis">
      Simplifying
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-one-locus-diploid-selection-in-discrete-time" class="md-nav__link">
    <span class="md-ellipsis">
      3. One-locus diploid selection in discrete time
    </span>
  </a>
  
    <nav class="md-nav" aria-label="3. One-locus diploid selection in discrete time">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#life-cycle-diagram_1" class="md-nav__link">
    <span class="md-ellipsis">
      Life-cycle diagram
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deriving-the-equations_1" class="md-nav__link">
    <span class="md-ellipsis">
      Deriving the equations
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#simplifying_1" class="md-nav__link">
    <span class="md-ellipsis">
      Simplifying
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-comparing-haploid-and-diploid-selection" class="md-nav__link">
    <span class="md-ellipsis">
      4. Comparing haploid and diploid selection
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-one-locus-haploid-selection-in-continuous-time" class="md-nav__link">
    <span class="md-ellipsis">
      5. One-locus haploid selection in continuous time
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6-comparing-haploid-selection-in-discrete-and-continuous-time" class="md-nav__link">
    <span class="md-ellipsis">
      6. Comparing haploid selection in discrete and continuous time
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-the-models-weve-covered" class="md-nav__link">
    <span class="md-ellipsis">
      7. The models we've covered
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<script type="text/x-thebe-config">
  {
      requestKernel: true,
      mountActivateWidget: true,
      mountStatusWidget: true,
      binderOptions: {
      repo: "mmosmond/executable-cells",
      ref: "main",
      },
  }
</script>
<script src="https://unpkg.com/thebe@latest/lib/index.js"></script>
<p><link rel="stylesheet" href="https://unpkg.com/thebe@latest/lib/thebe.css"></p>
<h1 id="lecture-4-one-locus-selection">Lecture 4: One-locus selection</h1>
<hr style="margin-bottom: 0em;">
<center>
<div class="inrow">
    Run notes interactively?
    <div style="float: left;" class="thebe-activate"></div>
    <div class="thebe-status"></div>
</div>
</center>
<hr style="margin-top: 0em;">

<h2 id="lecture-overview">Lecture overview</h2>
<ol>
<li><a href="#section1">Review of Hardy-Weinberg</a></li>
<li><a href="#section2">One-locus haploid selection in discrete time</a></li>
<li><a href="#section3">One-locus diploid selection in discrete time</a></li>
<li><a href="#section4">Comparing haploid and diploid selection</a></li>
<li><a href="#section5">One-locus haploid selection in continuous time</a></li>
<li><a href="#section6">Comparing haploid selection in discrete and continuous time</a></li>
<li><a href="#section7">The models we've covered</a></li>
</ol>
<p><span id='section1'></span></p>
<h2 id="1-review-of-hardy-weinberg">1. Review of Hardy-Weinberg</h2>
<hr>

<p>In Lecture 2 we saw that one round of random union among gametes (equivalently, random mating among diploids) causes the frequency of diploid genotypes at a locus with two alleles, <span class="arithmatex">\(A\)</span> and <span class="arithmatex">\(a\)</span>, to become</p>
<ul>
<li><span class="arithmatex">\(AA\)</span>: <span class="arithmatex">\(p^2\)</span></li>
<li><span class="arithmatex">\(Aa\)</span>: <span class="arithmatex">\(2pq\)</span></li>
<li><span class="arithmatex">\(aa\)</span>: <span class="arithmatex">\(q^2\)</span></li>
</ul>
<p>where <span class="arithmatex">\(p\)</span> is the frequency of allele <span class="arithmatex">\(A\)</span> and <span class="arithmatex">\(q=1-p\)</span> is the frequency of allele <span class="arithmatex">\(a\)</span>. </p>
<p>A population with these diploid genotype frequencies is said to be in Hardy-Weinberg equilibrium. Furthermore, we showed that under this model the allele frequencies do not change, <span class="arithmatex">\(p' = p\)</span>. </p>
<p>But what if there is natural selection?</p>
<p><span id='section2'></span></p>
<h2 id="2-one-locus-haploid-selection-in-discrete-time">2. One-locus haploid selection in discrete time</h2>
<hr>

<p>We begin by examining a model where selection acts during the haploid phase of the life-cycle. This is a little simpler than diploid selection because there are only two haploid genotypes, <span class="arithmatex">\(A\)</span> and <span class="arithmatex">\(a\)</span>. It is also a very relevant model for species with long haploid phases (eg, some algae and fungi) or with strong competition during the haploid phase (eg, pollen competiting for ovules).</p>
<h3 id="life-cycle-diagram">Life-cycle diagram</h3>
<p>We first illustrate the structure of the model with a life-cycle diagram.</p>
<center>
<pre class="mermaid"><code>    graph LR;
    A((p)) --gamete union--&gt; B((p'));
    B --meiosis--&gt; C((p''));
    C --selection--&gt; A;</code></pre>
</center>

<p>We will census the population at the beginning of the haploid phase (immediately after meiosis).</p>
<h3 id="deriving-the-equations">Deriving the equations</h3>
<p>Let the number of haploid individuals with each allele be</p>
<ul>
<li><span class="arithmatex">\(n_A(t) =\)</span> number of individuals with the <span class="arithmatex">\(A\)</span> allele in generation <span class="arithmatex">\(t\)</span></li>
<li><span class="arithmatex">\(n_a(t) =\)</span> number of individuals with the <span class="arithmatex">\(a\)</span> allele in generation <span class="arithmatex">\(t\)</span></li>
</ul>
<p>The frequency of <span class="arithmatex">\(A\)</span> is therefore <span class="arithmatex">\(p(t) = \frac{n_a(t)}{n_A(t) + n_a(t)}\)</span>.</p>
<p>Now, let’s assume that during selection each haploid individual has reproductive factor</p>
<ul>
<li><span class="arithmatex">\(W_A =\)</span> reproductive factor of individuals with the <span class="arithmatex">\(A\)</span> allele </li>
<li><span class="arithmatex">\(W_a =\)</span> reproductive factor of individuals with the <span class="arithmatex">\(a\)</span> allele</li>
</ul>
<p>These reproductive factors are referred to as the <strong>absolute fitnesses</strong> as they determine the (absolute) numbers of individuals after selection, <span class="arithmatex">\(n_i' = W_i n_i(t)\)</span> for <span class="arithmatex">\(i=A\)</span> and <span class="arithmatex">\(i=a\)</span>. To relate back to the models of population growth in Lecture 3, this is exponential growth of each allele.</p>
<p>After selection these alleles randomly pair, go through the dipliod phase of the life cycle, and then segregate back into haploids after meiosis. From our analysis of Hardy-Weinberg we know random mating and segregation don't affect allele frequency. The frequency of <span class="arithmatex">\(A\)</span> in the next generation is therefore</p>
<div class="arithmatex">\[
\begin{aligned}
p(t+1) 
&amp;= \frac{n_A(t+1)}{n_A(t+1) + v_a n_a(t)} \\
&amp;= \frac{W_A n_A(t)}{W_A n_A(t) + W_a n_a(t)}.
\end{aligned}
\]</div>
<p>To make this a recursion equation we need to write <span class="arithmatex">\(p(t+1)\)</span> in terms of <span class="arithmatex">\(p(t)\)</span>, so that if we knew <span class="arithmatex">\(p\)</span> at some point in time we can recursively calculate it in all future times. To do this we divide both the numerator and denominator by the total number of individuals, <span class="arithmatex">\(n_A(t) + n_a(t)\)</span>, and simplify</p>
<div class="arithmatex">\[
\begin{aligned}
p(t+1) 
&amp;= \frac{W_A n_A(t)}{W_A n_A(t) + W_a n_a(t)} \\
&amp;= \frac{W_A\frac{n_A(t)}{n_A(t) + n_a(t)}}{W_A\frac{n_A(t)}{n_A(t) + n_a(t)} + W_a\frac{n_a(t)}{n_A(t) + n_a(t)}}\\
&amp;= \frac{W_A p(t)}{W_A p(t) + W_a q(t)}.
\end{aligned}
\]</div>
<p>This is now a recursion equation for the allele frequency in our model of haploid selection.</p>
<h3 id="simplifying">Simplifying</h3>
<p>Our current recursion is a function of two parameters, the absolute fitnesses <span class="arithmatex">\(W_A\)</span> and <span class="arithmatex">\(W_a\)</span>. Now notice that if we divide both the numerator and denominator by one of these fitnesses, say <span class="arithmatex">\(W_a\)</span>, then we can reduce the recursion to a function of only one parameter, <span class="arithmatex">\(w_A = W_A/W_a\)</span>, the fitness of <span class="arithmatex">\(A\)</span> <em>relative</em> to the fitness of <span class="arithmatex">\(a\)</span>,</p>
<div class="arithmatex">\[
\begin{aligned}
p(t+1) &amp;= \frac{(W_A/W_a)p(t)}{(W_A/W_a)p(t) + (W_a/W_a)q(t)}\\
&amp;= \frac{w_A p(t)}{w_A p(t) + q(t)}.
\end{aligned}
\]</div>
<p>Note that the allele frequency dynamics depend only on the <strong>relative fitnesses</strong>, and not on the absolute fitnesses, meaning that evolution does not depend on how the size of the population changes. It is therefore possible to study evolution while ignoring population dynamics under this simple model.</p>
<p>Let's plot the recursion to get a sense of the dynamics.</p>
<pre data-executable="true" data-language="python">
import matplotlib.pyplot as plt

# calculate allele frequency over time with recursion
w_A = 1.1 #relative fitness of allele A
p_now = 0.01 #initial allele frequency, p_0
ps = [] #list to hold allele frequencies over time
ts = range(100) #list of time steps
for t in ts: #for each time
    ps.append(p_now) #add current allele frequency to list
    p_now = w_A*p_now/(w_A*p_now+(1-p_now)) #update allele frequency with our recursion equation

# plot
plt.scatter(ts, ps) #plot the (t,p) pairs
plt.xlabel("generation, $t$") #label axes
plt.ylabel("frequency of $A$ allele, $p(t)$")
plt.show()
</pre>

<p><img alt="png" src="../lecture-04_files/lecture-04_3_0.png" /></p>
<p><span id='section3'></span></p>
<h2 id="3-one-locus-diploid-selection-in-discrete-time">3. One-locus diploid selection in discrete time</h2>
<hr>

<p>Since we are all currently in the diploid phase of our life-cycle, it is natural to ask: <em>Does selection in the diploid phase work the same way?</em></p>
<h3 id="life-cycle-diagram_1">Life-cycle diagram</h3>
<p>We now have the following life-cycle diagram.</p>
<center>
<pre class="mermaid"><code>    graph LR;
    A((p)) --gamete union--&gt; B((p'));
    B --selection--&gt; C((p''));
    C --meiosis--&gt; A;</code></pre>
</center>

<p>We will census at the beginning of the diploid phase (immediately after gamete union).</p>
<h3 id="deriving-the-equations_1">Deriving the equations</h3>
<p>Let the number of diploid individuals with each genotype be</p>
<ul>
<li><span class="arithmatex">\(n_{AA}(t) =\)</span> number of individuals with the <span class="arithmatex">\(AA\)</span> genotype in generation <span class="arithmatex">\(t\)</span></li>
<li><span class="arithmatex">\(n_{Aa}(t) =\)</span> number of individuals with the <span class="arithmatex">\(Aa\)</span> genotype in generation <span class="arithmatex">\(t\)</span></li>
<li><span class="arithmatex">\(n_{aa}(t) =\)</span> number of individuals wite the <span class="arithmatex">\(aa\)</span> genotype in generation <span class="arithmatex">\(t\)</span></li>
</ul>
<p>The frequency of allele <span class="arithmatex">\(A\)</span> is then</p>
<div class="arithmatex">\[
\begin{aligned}
p(t)
&amp;= \frac{2n_{AA}(t) + n_{Aa}(t)}{2n_{AA}(t) + n_{Aa}(t) + n_{aa}(t)} \\
&amp;= \frac{n_{AA}(t) + \frac{1}{2}n_{Aa}(t)}{n_{AA}(t) + n_{Aa}(t) + n_{aa}(t)}
\end{aligned}
\]</div>
<p>Now, let’s assume that during selection each diploid individual has reproductive factor</p>
<ul>
<li><span class="arithmatex">\(W_{AA} =\)</span> reproductive factor of individuals with the <span class="arithmatex">\(AA\)</span> genotype </li>
<li><span class="arithmatex">\(W_{Aa} =\)</span> reproductive factor of individuals with the <span class="arithmatex">\(Aa\)</span> genotype</li>
<li><span class="arithmatex">\(W_{aa} =\)</span> reproductive factor of individuals with the <span class="arithmatex">\(aa\)</span> genotype</li>
</ul>
<p>These reproductive factors are again referred to as the <strong>absolute fitnesses</strong> as they determine the (absolute) numbers of individuals after selection, <span class="arithmatex">\(n_i' = W_i n_i(t)\)</span> for <span class="arithmatex">\(i=AA\)</span>, <span class="arithmatex">\(i=Aa\)</span>, and <span class="arithmatex">\(i=aa\)</span>.</p>
<p>After selection these genotypes segregate into haploids via meiosis, go through the haploid phase of the life cycle, and then random join to create diploids again. Again, we know random union and segregation don't affect allele frequency. The frequency of <span class="arithmatex">\(A\)</span> in the next generation is therefore</p>
<div class="arithmatex">\[
\begin{aligned}
p(t+1) 
&amp;= \frac{n_{AA}(t+1) + \frac{1}{2}n_{Aa}(t+1)}{n_{AA}(t+1) + n_{Aa}(t+1) + n_{aa}(t+1)}\\
&amp;= \frac{W_{AA}n_{AA}(t) + \frac{1}{2}W_{Aa}n_{Aa}(t)}{W_{AA}n_{AA}(t) + \frac{1}{2}W_{Aa}n_{Aa}(t) + W_{aa}n_{aa}(t)}.
\end{aligned}
\]</div>
<p>As above, we want a recursion equation in terms of allele frequency, so we want to replace the <span class="arithmatex">\(n_i\)</span>'s in the right hand side of this equation with <span class="arithmatex">\(p\)</span>'s. To do this we note that with the random union of gametes the diploid offspring are in Hardy-Weinberg proportions, so that</p>
<div class="arithmatex">\[
\begin{aligned}
n_{AA}(t) &amp;= p(t)^2 n(t) \\
n_{Aa}(t) &amp;= 2p(t) q(t) n(t) \\
n_{aa}(t) &amp;= q(t)^2 n(t)
\end{aligned}
\]</div>
<p>where <span class="arithmatex">\(n(t) = n_{AA}(t) + n_{Aa}(t) + n_{aa}(t)\)</span> is the total population size.</p>
<p>Substituting these Hardy-Weinberg proportions in and simplifying, the total population size cancels out and we can rewrite the above equation in terms of allele frequency alone,</p>
<div class="arithmatex">\[
\begin{aligned}
p(t+1) 
&amp;= \frac{W_{AA}p(t)^2 n(t) + W_{Aa}p(t) q(t) n(t)}{W_{AA}p(t)^2 n(t) + 2W_{Aa}p(t) q(t) n(t) + W_{aa}q(t)^2 n(t)}\\
&amp;= \frac{W_{AA}p(t)^2 + W_{Aa}p(t) q(t)}{W_{AA}p(t)^2 + 2W_{Aa}p(t) q(t) + W_{aa}q(t)^2}.
\end{aligned}
\]</div>
<p>This is a recursion equation for allele frequency in our model of diploid selection.</p>
<h3 id="simplifying_1">Simplifying</h3>
<p>As in the haploid selection case, we can divide by one of the absolute fitnesses, say <span class="arithmatex">\(W_{aa}\)</span>, to remove one parameter from the model</p>
<div class="arithmatex">\[
\begin{aligned}
p(t+1)
&amp;= \frac{(W_{AA}/W_{aa})p(t)^2 + (W_{Aa}/W_{aa})p(t) q(t)}{(W_{AA}/W_{aa})p(t)^2 + 2(W_{Aa}/W_{aa})p(t) q(t) + (W_{aa}/W_{aa})q(t)^2} \\
&amp;= \frac{w_{AA}p(t)^2 + w_{Aa}p(t) q(t)}{w_{AA}p(t)^2 + 2w_{Aa}p(t) q(t) + q(t)^2}
\end{aligned}
\]</div>
<p>This recursion is a function of only two relative fitnesses, <span class="arithmatex">\(w_{AA} = W_{AA}/W_{aa}\)</span> and <span class="arithmatex">\(w_{Aa} = W_{Aa}/W_{aa}\)</span>.</p>
<p>Let's plot the recursion to get a sense of the dynamics.</p>
<pre data-executable="true" data-language="python">
import matplotlib.pyplot as plt

# calculate allele frequency over time with recursion
w_AA = 1.2 #relative fitness of genotype AA
w_Aa = 1.1 #relative fitness of genotype Aa 
p_now = 0.01 #initial allele frequency, p_0
ps = [] #list to hold allele frequencies over time
ts = range(100) #list of time steps
for t in ts: #for each time
    ps.append(p_now) #add current allele frequency to list
    p_now = (w_AA*p_now**2 + w_Aa*p_now*(1-p_now))/(w_AA*p_now**2 + 2*w_Aa*p_now*(1-p_now) + (1-p_now)**2) #update allele frequency with our recursion equation

# plot
plt.scatter(ts, ps)
plt.xlabel("generation, $t$")
plt.ylabel("frequency of $A$ allele, $p(t)$")
plt.show()
</pre>

<p><img alt="png" src="../lecture-04_files/lecture-04_5_0.png" /></p>
<p><span id='section4'></span></p>
<h2 id="4-comparing-haploid-and-diploid-selection">4. Comparing haploid and diploid selection</h2>
<hr>

<p>So, returning to our original question, let's compare evolution under haploid selection</p>
<div class="arithmatex">\[
p(t+1) = \frac{W_A p(t)}{W_A p(t) + W_a q(t)}
\]</div>
<p>to evolution under diploid selection</p>
<div class="arithmatex">\[
p(t+1) = \frac{W_{AA}p(t)^2 + W_{Aa}p(t) q(t)}{W_{AA}p(t)^2 + 2W_{Aa}p(t) q(t) + W_{aa}q(t)^2}.
\]</div>
<p>To facilitate this, let’s assume the fitness of a diploid genotype is the product of the haploid fitnesses, i.e., <span class="arithmatex">\(W_{AA} = W_A W_A\)</span>, <span class="arithmatex">\(W_{Aa} = W_A W_a\)</span>, and <span class="arithmatex">\(W_{aa} = W_a W_a\)</span>.</p>
<p>It then happens that our diploid recursion reduces to the haploid recursion,</p>
<div class="arithmatex">\[
\begin{aligned}
p(t+1) &amp;= 
\frac{W_A W_A p(t)^2 + W_A W_a p(t) q(t)}{W_A W_A p(t)^2 + 2W_A W_a p(t) q(t) + W_a W_a q(t)^2}\\
&amp;= \frac{W_A p(t)(p(t) + W_a q(t))}{W_A p(t)(W_A p(t) + W_a q(t)) + W_a q(t)(W_A p(t) + W_a q(t))}\\
&amp;= \frac{W_A p(t)}{W_A p(t) + W_a q(t)}
\end{aligned}
\]</div>
<p>This shows that we need twice as much selection under diploid selection relative to that under haploid selection (e.g., <span class="arithmatex">\(W_{AA} = W_A W_A\)</span>) for evolution to proceed as quickly. <em>Why is evolution slower under diploid selection?</em></p>
<p>Let's check this visually (try breaking the assumption above, <span class="arithmatex">\(W_{ij}=W_i W_j\)</span>, to see what happens).</p>
<pre data-executable="true" data-language="python">
import matplotlib.pyplot as plt

# calculate allele frequency over time with recursion
w_A = 1.1 #relative fitness of allele A in the haploid model
w_AA = w_A*w_A #relative fitness of genotype AA in the diploid model
w_Aa = w_A #relative fitness of genotype Aa in the diploid model
p_now_hap = 0.01 #initial allele frequency, p_0
p_now_dip = p_now_hap
ps_hap, ps_dip = [], [] #lists to hold allele frequencies over time
ts = range(100) #list of time steps
for t in ts: #for each time
    ps_hap.append(p_now_hap) #add current allele frequency to list
    p_now_hap = w_A*p_now_hap/(w_A*p_now_hap+(1-p_now_hap)) #update allele frequency with haploid recursion equation
    ps_dip.append(p_now_dip) #add current allele frequency to list
    p_now_dip = (w_AA*p_now_dip**2 + w_Aa*p_now_dip*(1-p_now_dip))/(w_AA*p_now_dip**2 + 2*w_Aa*p_now_dip*(1-p_now_dip) + (1-p_now_dip)**2) #update allele frequency with diploid recursion equation

# plot
plt.scatter(ts, ps_hap, label='haploid')
plt.scatter(ts, ps_dip, s=10, label='diploid')
plt.xlabel("generation, $t$")
plt.ylabel("frequency of $A$ allele, $p(t)$")
plt.legend() #add legend to know which points belong to which model
plt.show()
</pre>

<p><img alt="png" src="../lecture-04_files/lecture-04_7_0.png" /></p>
<p><span id='section5'></span></p>
<h2 id="5-one-locus-haploid-selection-in-continuous-time">5. One-locus haploid selection in continuous time</h2>
<hr>

<p>To model haploid selection in continuous-time we assume that during the selective phase each haploid individual has growth rate</p>
<ul>
<li><span class="arithmatex">\(r_A =\)</span> growth rate of individuals with the <span class="arithmatex">\(A\)</span> allele</li>
<li><span class="arithmatex">\(r_a =\)</span> growth rate of individuals with the <span class="arithmatex">\(a\)</span> allele</li>
</ul>
<p>and the relative numbers of each type don't change otherwise (ie, during union, diploidy, or meiosis).</p>
<p>We therefore have exponential growth of both genotypes, <span class="arithmatex">\(\frac{\mathrm{d} n_i}{dt} = r_i n_i\)</span>.</p>
<p>At any particular point in time, <span class="arithmatex">\(t\)</span>, the frequency of allele <span class="arithmatex">\(A\)</span> is, <span class="arithmatex">\(p(t) = n_A(t)/(n_A(t) + n_a(t))\)</span>. </p>
<p>We can therefore derive the rate of change in the frequency of allele <span class="arithmatex">\(A\)</span>, <span class="arithmatex">\(\mathrm{d}p/\mathrm{d}t\)</span>, using the qoutient rule (see Appendix 2 in the text for help with this and other math tricks)</p>
<div class="arithmatex">\[
\begin{aligned}
\frac{\mathrm{d}p}{\mathrm{d}t}
&amp;= \frac{\mathrm{d}\frac{n_A}{n_A + n_a}}{\mathrm{d}t} \\
&amp;= \frac{\frac{\mathrm{d}n_A}{\mathrm{d}t} (n_A(t) + n_a(t)) - n_A(t) \frac{\mathrm{d}(n_A+n_a)}{\mathrm{d}t}}{(n_A(t) + n_a(t))^2}\\
&amp;= \frac{r_A n_A(t) (n_A(t) + n_a(t)) - n_A(t) (r_A n_A(t) + r_a n_a(t))}{(n_A(t) + n_a(t))^2}\\
&amp;= \frac{r_A n_A(t) n_a(t) - r_a n_A(t) n_a(t))}{(n_A(t) + n_a(t))^2}\\
&amp;= \frac{(r_A -r_a)n_A(t) n_a(t)}{(n_A(t) + n_a(t))^2}\\
&amp;= (r_A -r_a)p(t) q(t),
\end{aligned}
\]</div>
<p>where <span class="arithmatex">\(s_c = r_A - r_a\)</span> is the continuous-time <strong>selection coefficient</strong> of allele <span class="arithmatex">\(A\)</span>.</p>
<p>A similar equation can be derived for the model of diploid-selection in continuous time, but we will not study it (see Problem 3.16 in the text if you are curious).</p>
<p><span id='section6'></span></p>
<h2 id="6-comparing-haploid-selection-in-discrete-and-continuous-time">6. Comparing haploid selection in discrete and continuous time</h2>
<hr>

<p>Are the discrete- and continuous-time models of haploid selection as different as they look?</p>
<div class="arithmatex">\[
p(t+1) = \frac{W_A p(t)}{W_A p(t) + W_a q(t)}
\]</div>
<div class="arithmatex">\[
\frac{\mathrm{d}p}{\mathrm{d}t} = s_c p(t) q(t)
\]</div>
<p>Not really. Discrete and continuous time models generally behave in a similar fashion when changes occur slowly over time. For this model of haploid selection, this implies that the discrete and continuous models will be similar when the fitnesses of the two alleles are nearly equal, i.e., when <span class="arithmatex">\(W_A - W_a\)</span> is small. This is called "weak selection".</p>
<p>In the discrete model, the change in the allele frequency is</p>
<div class="arithmatex">\[
\begin{aligned}
\Delta p &amp;= p(t+1) - p(t) \\
&amp;= \frac{W_A p(t)}{W_A p(t) + W_a q(t)} - p(t) \\
&amp;= \frac{(W_A - W_a)p(t) q(t)}{W_A p(t) + W_a q(t)}.
\end{aligned}
\]</div>
<p>Now define <span class="arithmatex">\(s_d = (W_A - W_a)/W_a\)</span> as the discrete time selection coefficient. Subbing this in as <span class="arithmatex">\(W_A = s_d W_a + W_a\)</span> gives</p>
<div class="arithmatex">\[
\begin{aligned}
\Delta p 
&amp;= \frac{(W_A-W_a)p(t) q(t)}{W_A p(t) + W_a q(t)}\\
&amp;= \frac{(s_d W_a + W_a - W_a) p(t) q(t)}{(s_d W_a + W_a) p(t) + W_a q(t)}\\
&amp;= \frac{s_d W_a p(t) q(t)}{s_d W_a p(t) + W_a (p(t) + q(t))}\\
&amp;= \frac{s_d W_a p(t) q(t)}{s_d W_a p(t) + W_a}\\
&amp;= \frac{s_d p(t) q(t)}{s_d p(t) + 1}\\
\end{aligned}
\]</div>
<p>Now assume weak selection, i.e., that <span class="arithmatex">\(s_d\)</span> is small. This implies <span class="arithmatex">\(s_d p(t) + 1 \approx 1\)</span>. Making this approximation we have</p>
<div class="arithmatex">\[
\Delta p \approx s_d p(t) q(t)
\]</div>
<p>This is equivalent to the continuous-time model when the selection coefficients are equal, <span class="arithmatex">\(s_c = s_d\)</span>.</p>
<p><span id='section7'></span></p>
<h2 id="7-the-models-weve-covered">7. The models we've covered</h2>
<hr>

<p>To summarize the last two lectures, we've derived four of the most classic models in ecology and evolution:</p>
<table>
<thead>
<tr>
<th>Model</th>
<th>Discrete time</th>
<th>Continous time</th>
</tr>
</thead>
<tbody>
<tr>
<td>Exponential growth</td>
<td><span class="arithmatex">\(n(t+1) = R n(t)\)</span></td>
<td><span class="arithmatex">\(\frac{\mathrm{d}n}{\mathrm{d}t} = r n(t)\)</span></td>
</tr>
<tr>
<td>Logistic growth</td>
<td><span class="arithmatex">\(n(t+1) = (1 + r(1 - \frac{n(t)}{K}))n(t)\)</span></td>
<td><span class="arithmatex">\(\frac{\mathrm{d}n}{\mathrm{d}t} = r(1 - \frac{n(t)}{K})n(t)\)</span></td>
</tr>
<tr>
<td>Haploid selection</td>
<td><span class="arithmatex">\(p(t+1) = \frac{W_A p(t)}{W_A p(t) + W_a q(t)}\)</span></td>
<td><span class="arithmatex">\(\frac{\mathrm{d}p}{\mathrm{d}t} = s p(t)(1-p(t))\)</span></td>
</tr>
<tr>
<td>Diploid selection</td>
<td><span class="arithmatex">\(p(t+1) = \frac{W_{AA}p(t)^2 + W_{Aa}p(t) q(t)}{W_{AA}p(t)^2 + 2 W_{Aa} p(t) q(t) + W_{aa} q(t)^2}\)</span></td>
<td>Not derived</td>
</tr>
</tbody>
</table>
<p>See textbook Sections 3.4 and 3.5 for models of interacting species and epidemiology, respectively, which we won't cover in class.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.annotate", "navigation.tabs"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.92b07e13.min.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>